File: abc137/abc137_c.cs
// https://atcoder.jp/contests/abc137/submissions/7962505

using System;
using System.Collections.Generic;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var dic = new Dictionary<string, int>();

            var result = 0L;
            for (int i = 0; i < N; i++)
            {
                var s = Console.ReadLine();

                var array = s.ToCharArray();
                Array.Sort(array);
                s = new string(array);

                if (dic.ContainsKey(s))
                {
                    dic[s]++;
                }
                else
                {
                    dic[s] = 0;
                }

                result += dic[s];
            }

            Console.WriteLine(result);
        }
    }
}

File: abc/abc137/abc137_c.fs
// https://atcoder.jp/contests/abc137/submissions/11991568

open System
open System.Collections.Generic

let N = stdin.ReadLine() |> int
let dic = new Dictionary<string, int64>()

let rec solver s i =
    if i = N then s
    else
        let rs =
            stdin.ReadLine().ToCharArray()
            |> Array.sort
            |> fun x -> new String(x)
        if dic.ContainsKey(rs) then dic.[rs] <- dic.[rs] + 1L
        else dic.[rs] <- 0L
        solver (s + dic.[rs]) (i + 1)

solver 0L 0 |> stdout.WriteLine
